lib/modules/manager/buildpacks/schema.ts

```diff
@@ -1,10 +1,33 @@
 import { z } from 'zod';
 import { Toml } from '../../../util/schema-utils';
 
-const BuildpackGroup = z.object({
-  uri: z.string().optional(),
+const BuildpackByName = z.object({
+  id: z.string(),
+  version: z.string().optional(),
 });
 
+const BuildpackByURI = z.object({
+  uri: z.string(),
+});
+
+const BuildpackGroup = BuildpackByName.or(BuildpackByURI);
+
+type BuildpackByNameType = z.infer<typeof BuildpackByName>;
+type BuildpackByURIType = z.infer<typeof BuildpackByURI>;
+type BuildpackGroupType = z.infer<typeof BuildpackGroup>;
```

Comment:
```suggestion
type BuildpackByName = z.infer<typeof BuildpackByName>;
type BuildpackByURI = z.infer<typeof BuildpackByURI>;
type BuildpackGroup = z.infer<typeof BuildpackGroup>;
```
move to below the defining const.

don't use type suffix, typescript is smart enough to handle it.